server/service/src/invoice/prescription/update/generate.rs

```diff
@@ -95,10 +96,30 @@ fn should_update_batches_total_number_of_packs(
     }
 }
 
+fn backdate_status_datetimes(invoice: &mut InvoiceRow, backdated_datetime: NaiveDateTime) {
+    match &invoice.status {
+        InvoiceStatus::New => {} // Created datetime is never backdated
+        InvoiceStatus::Allocated => {
+            invoice.allocated_datetime = Some(backdated_datetime);
+        }
+        InvoiceStatus::Picked => {
+            invoice.allocated_datetime = Some(backdated_datetime);
```

Comment:
Is this safe, should we always have an allocated date? Maybe we should check if it's not null before setting it?

Reply:
I find it easer to think about if we first do normal update of statuses, and then adjust them by backdate (for those dates that are set)

https://github.com/msupply-foundation/open-msupply/compare/4900-backdated_datetime...4900-backdated_datetime(PR-review-suggestion)